suomen-antropologi-journal-of-the-finnish-anthropological-society.csl

```diff
@@ -0,0 +1,634 @@
+        <group delimiter=": ">
+          <text variable="publisher-place"/>
+          <text variable="publisher"/>
+        </group>
+      </if>
+      <else-if type="thesis" match="any">
+        <group delimiter=": ">
+          <text variable="publisher-place"/>
+          <text variable="publisher" suffix=" dissertation"/>
+        </group>
+      </else-if>
+      <else-if type="post-weblog webpage" match="none">
+        <group delimiter=", ">
+          <choose>
+            <if variable="event version" type="speech motion_picture" match="none">
+              <text variable="genre"/>
+            </if>
+          </choose>
+          <choose>
+            <if type="article-journal article-magazine" match="none">
+              <group delimiter=": ">
+                <choose>
+                  <if variable="publisher-place">
+                    <text variable="publisher-place"/>
+                  </if>
+                  <else>
+                    <text variable="event-place"/>
+                  </else>
+                </choose>
+                <text variable="publisher"/>
+              </group>
+            </if>
+          </choose>
+        </group>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="event">
+    <choose>
+      <if variable="container-title" match="none">
+        <choose>
+          <if variable="event">
+            <choose>
+              <if variable="genre" match="none">
+                <text term="presented at" text-case="capitalize-first" suffix=" "/>
+                <text variable="event"/>
+              </if>
+              <else>
+                <group delimiter=" ">
+                  <text variable="genre" text-case="capitalize-first"/>
+                  <text term="presented at"/>
+                  <text variable="event"/>
+                </group>
+              </else>
+            </choose>
+          </if>
+          <else-if type="speech">
+            <text variable="genre" text-case="capitalize-first"/>
+          </else-if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="issued">
+    <choose>
+      <if type="bill legal_case legislation" match="none">
+        <choose>
+          <if variable="issued">
+            <group prefix=" ">
+              <date variable="issued">
+                <date-part name="year"/>
+              </date>
+              <text variable="year-suffix"/>
+              <choose>
+                <if type="speech" match="any">
+                  <date variable="issued">
+                    <date-part prefix=", " name="month"/>
+                  </date>
+                </if>
+                <else-if type="article-journal bill book chapter graphic legal_case legislation motion_picture paper-conference report song dataset" match="none">
+                  <date form="numeric" variable="issued">
+                    <date-part name="month"/>
+                    <date-part name="day"/>
+                  </date>
```

Comment:
Setting `form` on a `<date>` element includes all of the date parts. To include only some date parts, `form` must be omitted from `<date>`. 

```suggestion
                  <date variable="issued">
                    <date-part name="month" form="numeric" prefix="-"/>
                    <date-part name="day" form="numeric" prefix="-"/>
                  </date>
```

I assume that you want dates in the form `2021-04-16`?

Reply:
> See the fix and question above.

Thanks @bwiernik , now I spotted it in the preview. Is there a possibility to move the day and month variables after the name of the publication in this situation? The preferred form is:
**Foderaro, Lisa W.** 2012. Rooftop Greenhouse Will Boost City Farming. _New York Times_ 6 April, p. A20. New York.

Sorry, I'm at the limits of my CSL skills. If moving the date-part variables is not possible, then lines 355-358 can be ommitted, i.e. the whole `<date variable="issued">` element, because we only need the publication year.

Reply:
So you want them completely separated, or would 2010, 6 April be fine?

Reply:
Yes, if it's possible in CSL to move the '<date variable="issued>' element, with only "6 April" (i.e. day + month) after the name of the publication, then that would be ideal. See bolded part.
**Foderaro, Lisa W.** 2012. Rooftop Greenhouse Will Boost City Farming. _New York Times_ **6 April**, p. A20. New York.

If that is not possible, then I will just omit the whole <date variable="issued"> element so that the outcome will be:
**Foderaro, Lisa W.** 2012. Rooftop Greenhouse Will Boost City Farming. _New York Times_, p. A20. New York.

Reply:
And not in bold right?

Reply:
Yes, that's right! The day (numeric) month (long) not in bold. (The author is bolded in the journal style.)

Reply:
Hi,
are you able to help me resolve the issue in the code? That would be greatly appreciated. Fixing it is unfortunately beyond my abilities.

Reply:
1. Remove this part from your `issued` macro:
```xml
<date form="numeric" variable="issued">
  <date-part name="month"/>
  <date-part name="day"/>
</date>
```
(Or, maybe just include the relevant item-types to your `choose` conditions...)

2. Make a new macro `issued-day-month` using the suggestion made by @bwiernik 
3. Call that macro somewhere, probably either in the macros `container` or `container-title`.

Reply:
Hi @denismaier and @bwiernik,

thanks for the help. I think I have now solved the issue. I removed the superfluous parts from the `issued` macro and called the date-parts day and month in the container title conditioned to appear only for articles, newspaper articles and magazine articles.

Thanks for your help!

Reply:
So, the output is now correct?

Reply:
Yes. It works now as intended.